moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -527,36 +542,32 @@ TwistCommand Servo::toPlanningFrame(const TwistCommand& command) const
   return TwistCommand{ servo_params_.planning_frame, transformed_twist };
 }
 
-PoseCommand Servo::toPlanningFrame(const PoseCommand& command) const
+PoseCommand Servo::toPlanningFrame(const PoseCommand& command)
 {
   return PoseCommand{ servo_params_.planning_frame,
                       getPlanningToCommandFrameTransform(command.frame_id) * command.pose };
 }
 
-KinematicState Servo::getCurrentRobotState() const
+KinematicState Servo::getCurrentRobotState()
 {
-  moveit::core::RobotStatePtr robot_state = planning_scene_monitor_->getStateMonitor()->getCurrentState();
-  const moveit::core::JointModelGroup* joint_model_group =
-      robot_state->getJointModelGroup(servo_params_.move_group_name);
-  const auto joint_names = joint_model_group->getActiveJointModelNames();
+  robot_state_ = planning_scene_monitor_->getStateMonitor()->getCurrentState();
 
-  KinematicState current_state(joint_names.size());
-  current_state.joint_names = joint_names;
-  robot_state->copyJointGroupPositions(joint_model_group, current_state.positions);
-  robot_state->copyJointGroupVelocities(joint_model_group, current_state.velocities);
-  robot_state->copyJointGroupAccelerations(joint_model_group, current_state.accelerations);
+  KinematicState current_state(joint_names_.size());
+  current_state.joint_names = joint_names_;
+  robot_state_->copyJointGroupPositions(joint_model_group_, current_state.positions);
+  robot_state_->copyJointGroupVelocities(joint_model_group_, current_state.velocities);
+  robot_state_->copyJointGroupAccelerations(joint_model_group_, current_state.accelerations);
 
   return current_state;
 }
```

Comment:
Yeah, so this function seems to have 2 uses right now, which is why it's not `const`. I think what we need is:

1. An `updateCurrentRobotState()` that does everything done here but doesn't return `current_state`, it simply updates `robot_state_`, to be used in functions that calculate commands. 
2. A `getCurrentRobotState()` function that is `const` and simply returns `robot_state_`, to be used in other tools.